packages/app/src/cli/services/dev/app-events/app-watcher-esbuild.ts

```diff
@@ -79,11 +90,36 @@ export class ESBuildContextManager {
   }
 
   async deleteContexts(extensions: ExtensionInstance[]) {
-    const promises = extensions.map((ext) => this.contexts[ext.handle]?.dispose())
+    const promises = extensions.map((ext) => this.contexts[ext.handle]?.map((context) => context.dispose())).flat()
     await Promise.all(promises)
     extensions.forEach((ext) => {
       const {[ext.handle]: _, ...rest} = this.contexts
       this.contexts = rest
     })
   }
+
+  private async extensionEsBuildOptions(
+    extension: ExtensionInstance,
+    overrides: Pick<BundleOptions, 'stdin'> & {outputFileName?: string},
+  ) {
+    const outputPath = overrides.outputFileName
+      ? joinPath(dirname(extension.outputPath), overrides.outputFileName)
+      : extension.getOutputPathForDirectory(this.outputPath)
```

Comment:
This is not correct, the output path for the non-main assets should be:

```ts
const mainOutputPath = extension.getOutputPathForDirectory(this.outputPath)
const assetsDirectory = dirname(mainOutputPath)
const assetOutputPath = joinPath(assetsDirectory, asset.outputFileName)
```

The way it is right now, doesn't work for Consistent-dev (it is storing each file in a different folder)

My recommendation is to remove the "overrides" concept, you should always pass options to the config:
```ts
  private async extensionEsBuildOptions(
    extension: ExtensionInstance,
    stdin: StdinOptions,
    outputFileName: string,
  )
```